src/EditorFeatures/CSharpTest2/Recommendations/WhenKeywordRecommenderTests.cs

```diff
@@ -9,67 +9,530 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case int.MinValue $$ break; }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterMemberAccess_BeforeWhen() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case int.MinValue $$ when }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterBinaryExpression() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case 1 + 1 $$ }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterBinaryExpression_BeforeBreak() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case 1 + 1 $$ break; }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterBinaryExpression_BeforeWhen() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case 1 + 1 $$ when }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterTernaryExpression() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case true ? 1 : 1 $$ }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterTernaryExpression_BeforeBreak() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case true ? 1 : 1 $$ break; }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterTernaryExpression_BeforeWhen() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case true ? 1 : 1 $$ when }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterParenthesesWithIncompleteExpressionInside() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case (1 + ) $$ }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterParenthesesWithIncompleteExpressionInside_BeforeBreak() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case (1 + ) $$ break; }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_AfterParenthesesWithIncompleteExpressionInside_BeforeWhen() =>
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case (1 + ) $$ when }"));
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestForSwitchCase_NotAfterIncompleteMemberAccess() =>
+            await VerifyAbsenceAsync(AddInsideMethod(@"switch (1) { case int.$$ }"));
```

Comment:
well... now I have an urge to change this to use the same pattern as I did in #25703 (use a theory called 'NotAfterIncompleteExpression' with all these cases as inline data)

Reply:
even though here there wouldn't be as big of a benefit since all these tests are just one line

Reply:
I quickly made the change, I hope it's fine